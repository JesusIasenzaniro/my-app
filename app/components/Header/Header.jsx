import React from 'react';
import styles from './Header.module.css';
import Link from 'next/link';

const Header = () => {
    return (
        <ul className={styles.header}>
            <Link href='/' className={styles.link} data-testid='link'>
                <li>Home</li>
            </Link>
        </ul>
    );
};

export default Header;
