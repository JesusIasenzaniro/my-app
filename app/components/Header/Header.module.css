.header {
    border-bottom: 1px solid cornflowerblue;
    padding: 1rem;
    list-style-type: none;
}

.header li {
    font-weight: bold;
    color: cornflowerblue;
}

.link {
    text-decoration: none;
}

.link:hover {
    text-decoration: underline;
}
