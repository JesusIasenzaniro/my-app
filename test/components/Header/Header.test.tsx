import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '@/app/components/Header/Header';
import renderer from 'react-test-renderer';
import Link from 'next/link';

describe('Header component', () => {
    it('Should render properly', () => {
        render(<Header />);

        const listMenu = screen.getByRole('list');

        expect(listMenu);
    });

    it('Should redirect the user to home with Link', () => {
        const tree = renderer.create(<Link href='/'></Link>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
