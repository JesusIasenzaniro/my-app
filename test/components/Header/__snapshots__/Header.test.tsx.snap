// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should redirect the user to home with Link 1`] = `
<a
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
/>
`;
